subgraphs/abracadabra/src/common/setters.ts

```diff
@@ -0,0 +1,88 @@
+
+export function updateProtocolMarketList(marketAddress: string): void {
+  let protocol = getOrCreateLendingProtocol();
+  let marketIdList = protocol.marketIdList;
+  marketIdList.push(marketAddress);
+  protocol.marketIdList = marketIdList;
+  protocol.save();
+}
+
+export function createMarket(marketAddress: string, blockNumber: BigInt, blockTimestamp: BigInt): void {
+  let MarketEntity = new Market(marketAddress);
+  let MarketContract = cauldron.bind(Address.fromString(marketAddress));
+  let collateralCall = MarketContract.try_collateral();
+  if (!collateralCall.reverted) {
+    let inputToken = getOrCreateToken(collateralCall.value);
+    MarketEntity.protocol = getOrCreateLendingProtocol().id;
+    MarketEntity.inputTokens = [inputToken.id];
+    MarketEntity.outputToken = MIM;
+    MarketEntity.totalValueLockedUSD = BIGDECIMAL_ZERO;
+    MarketEntity.totalVolumeUSD = BIGDECIMAL_ZERO;
+    MarketEntity.inputTokenBalances = [BIGINT_ZERO];
+    MarketEntity.outputToken = getOrCreateToken(Address.fromString(MIM)).id;
+    MarketEntity.outputTokenSupply = BIGINT_ZERO;
+    MarketEntity.outputTokenPriceUSD = BIGDECIMAL_ZERO;
+    MarketEntity.createdTimestamp = blockTimestamp;
+    MarketEntity.createdBlockNumber = blockNumber;
+    MarketEntity.name = inputToken.name + " Market";
+    MarketEntity.isActive = true;
+    MarketEntity.canUseAsCollateral = true;
+    MarketEntity.canBorrowFrom = true;
+
+    if (marketAddress.toLowerCase() == "0x551a7cff4de931f32893c928bbc3d25bf1fc5147".toLowerCase()) {
+      MarketEntity.maximumLTV = bigIntToBigDecimal(BigInt.fromI32(90000), 5);
+      MarketEntity.liquidationPenalty = bigIntToBigDecimal(BigInt.fromI32(103000), 5).minus(BIGDECIMAL_ONE);
+      MarketEntity.liquidationThreshold = bigIntToBigDecimal(BigInt.fromI32(75000), 5); // ???
+      MarketEntity.stableBorrowRate = bigIntToBigDecimal(BigInt.fromI32(253509908), 18).times(SECONDS_PER_YEAR);
+    } else if (marketAddress.toLowerCase() == "0x6Ff9061bB8f97d948942cEF376d98b51fA38B91f".toLowerCase()) {
+      MarketEntity.maximumLTV = bigIntToBigDecimal(BigInt.fromI32(75000), 5);
+      MarketEntity.liquidationPenalty = bigIntToBigDecimal(BigInt.fromI32(112500), 5).minus(BIGDECIMAL_ONE);
+      MarketEntity.liquidationThreshold = bigIntToBigDecimal(BigInt.fromI32(75000), 5); // ???
+      MarketEntity.stableBorrowRate = bigIntToBigDecimal(BigInt.fromI32(475331078), 18).times(SECONDS_PER_YEAR);
+    } else if (marketAddress.toLowerCase() == "0xffbf4892822e0d552cff317f65e1ee7b5d3d9ae6".toLowerCase()) {
+      MarketEntity.maximumLTV = bigIntToBigDecimal(BigInt.fromI32(75000), 5);
+      MarketEntity.liquidationPenalty = bigIntToBigDecimal(BigInt.fromI32(112500), 5).minus(BIGDECIMAL_ONE);
+      MarketEntity.liquidationThreshold = bigIntToBigDecimal(BigInt.fromI32(75000), 5); // ???
+      MarketEntity.stableBorrowRate = bigIntToBigDecimal(BigInt.fromI32(475331078), 18).times(SECONDS_PER_YEAR);
```

Comment:
Can the rates change or are they hardcoded in the smart contract?

Reply:
rates are hard coded into the contract